src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,124 @@
+    private int quantity = DEFAULT_SIZE;
+    private Object[] elements = new Object[DEFAULT_CAPACITY];
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        elements[quantity] = value;
+        quantity += 1;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkInRangeForAddition(index);
+        growIfArrayFull();
+        Object [] temp = new Object[elements.length];
+        System.arraycopy(elements, 0, temp, 0, temp.length);
+        for (int i = 0, j = 0; i <= quantity; i++, j++) {
+            if (i == index) {
+                elements[i] = value;
+                i++;
+            }
+            elements[i] = temp[j];
+        }
+        quantity += 1;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkInRangeForGet(index);
+        return (T) (elements[index]);
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkInRangeForGet(index);
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkInRangeForGet(index);
+        Object [] temp = new Object[elements.length];
+        System.arraycopy(elements, 0, temp, 0, elements.length);
+        for (int i = 0, j = 0; i < quantity; i++, j++) {
+            if (i == index) {
+                j++;
+            }
+            if (j != quantity) {
+                elements[i] = temp[j];
+            }
+        }
+        quantity -= 1;
+        elements[quantity] = null;
+        return (T) (temp[index]);
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < quantity; i++) {
+            if (elements[i] == null && element == null) {
+                return remove(i);
+            }
+            if (element != null && elements[i] == null) {
+                continue;
+            }
+            if (elements[i].equals(element)) {
+                return remove(i);
+            }
+        }
+        throw new NoSuchElementException("No such element present");
     }
 
     @Override
     public int size() {
-        return 0;
+        return quantity;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size() == 0;
+    }
+
+    private void growIfArrayFull() {
+        if (size() == elements.length) {
+            int index = 0;
+            if (elements.length % 2 == 0) {
+                index = elements.length + elements.length / 2;
+            } else {
+                index = elements.length + elements.length / 2 + 1;
+            }
+            Object[] temp = new Object[index];
+            for (int i = 0; i < elements.length; i++) {
+                temp[i] = elements[i];
+            }
+            elements = temp;
+        }
```

Comment:
The resize logic can be simplified and made more efficient. Instead of creating a new array and manually copying all elements from the old array to the new one, you can use System.arraycopy(). Also, the new capacity can be calculated more simply as 'elements.length + elements.length / 2 + 1'. This will make your code more efficient and easier to read.